packages/@sanity/schema/src/legacy/types/file.ts

```diff
@@ -31,7 +38,7 @@ export const FileType = {
   extend(rawSubTypeDef: any, createMemberType: any) {
     const options = {...(rawSubTypeDef.options || DEFAULT_OPTIONS)}
 
-    const fields = [ASSET_FIELD, ...(rawSubTypeDef.fields || [])]
+    const fields = [ASSET_FIELD, MEDIA_LIBRARY_ASSET_FIELD, ...(rawSubTypeDef.fields || [])]
```

Comment:
We don't happen to have access to the sanity config here? IWBN if we could only include this field if `mediaLibrary.enabled == true`

Reply:
Is there a world where we could pass a flag to Schema.compile ?

Reply:
> You are probably right. I will need a trace in extract tho (if you missed the pr comment)

Working on it @snorrees :)